### PR TITLE
Add support for capturing passthrough arguments

### PR DESCRIPTION
**Pull request by pbelbin**
closes https://github.com/sparksuite/waterfall-cli/issues/276

Adds to the `.spec.ts` file, the optional `passThrough: true` option, and, to the `InputObject`, an optional array (`passThrough`) holding the captured arguments being passed through.

The passThrough items are in the same sequence as provided on the command line.

This allows users to do something like:

`utilCommand --option one -- --extra things here`

The `--extra things here` will appear in the response object from `waterfall.parse()` as: `.passThrough: [ '--extra', 'things', 'here']` and it's up to the code then, to do whatever is appropriate.

If support for passThrough is not enabled, and it's used, the user will be provided with an error message informing them of this situation.

Here is a screen shot of how a client app, not having the `passThrough` enabled for the 'run' command indicates improper usage:

Image 2020-07-24 at 2 27 50 PM

### CODE REVIEW

**Comment by pbelbin** [comment]
I noticed that there was no type information for the result of waterfall.parse(), so I added it.

File: src/utils.ts [hunk 1]
```diff
@@ -191,6 +208,18 @@ export default function utils(currentSettings: Settings) {
 					continue;
 				}
 
+				// Detect pass-through indication
+				if (argument === '--') {
+					// Error if this command does not accept pass-through arguments
+					if (!mergedSpec.passThrough) {
+						throw new ErrorWithoutStack(`Unrecognized argument: ${argument}`);
```
Comment (WesCossick) [inline src/utils.ts:215]: This error should be more helpful. Try:

```suggestion
						throw new ErrorWithoutStack('This command does not support pass-through arguments');
```

**Comment by WesCossick** [comment]
Can you add a screenshot of the "This command does not support pass-through arguments" error to the PR's description?

File: test/utils.test.ts [hunk 2]
```diff
@@ -744,6 +744,32 @@ describe('#organizeArguments()', () => {
 			utils(settings).organizeArguments();
 		}).toThrow();
 	});
+
+	test('Complains about unexpected pass-through arguments ', () => {
+		const settings = {
+			...settingsPizzaOrdering,
+			arguments: ['order', 'dine-in', '--', '--passThrough-flag', 'passThroughArgument1', 'passThroughArgument2'],
```
Comment (WesCossick) [inline test/utils.test.ts:751]:
```suggestion
			arguments: ['order', 'dine-in', '--', '--pass-through-flag', '--pass-through-option=true', 'pass-through-data'],
```

...can you update these arguments in the other two places, too?

**Comment by pbelbin** [comment]
> Can you also bump the minor version?

Done.

**Comment by WesCossick** [comment]
It doesn't look like the minor version has been bumped. See: https://semver.org/.

**Comment by pbelbin** [comment]
> It doesn't look like the minor version has been bumped. See: https://semver.org/.

Yeah, I bumped the wrong value.  Fixed now.

**Comment by pbelbin** [comment]
> Can you add a screenshot of the "This command does not support pass-through arguments" error to the PR's description?

Done.

**Comment by pbelbin** [comment]
> I noticed that there was no type information for the result of waterfall.parse(), so I added it.

I backed off on this because the impact of this to clients is bigger than what I want to bit off right now.

File: src/types.ts [hunk 3]
```diff
@@ -31,12 +31,14 @@ export interface CommandSpec {
 		};
 	};
 	executeOnCascade?: true;
+	passThrough?: true;
 }
 
 export interface InputObject {
 	command: string;
 	data?: string | number;
-	[index: string]: boolean | string | number | undefined;
```
Comment (WesCossick) [inline src/types.ts:39]: Why was this line changed?

File: test/utils.test.ts [hunk 4]
```diff
@@ -744,6 +744,32 @@ describe('#organizeArguments()', () => {
 			utils(settings).organizeArguments();
 		}).toThrow();
 	});
+
+	test('Complains about unexpected pass-through arguments ', () => {
+		const settings = {
+			...settingsPizzaOrdering,
+			arguments: ['order', 'dine-in', '--', '--pass-through-flag', 'pass-through-option=true', 'pass-through-data'],
```
Comment (WesCossick) [inline test/utils.test.ts:751]: This is a better representation of what an option is:

```suggestion
			arguments: ['order', 'dine-in', '--', '--pass-through-flag', 'pass-through-option=value', 'pass-through-data'],
```

File: test/utils.test.ts [hunk 5]
```diff
@@ -744,6 +744,32 @@ describe('#organizeArguments()', () => {
...
+
+		expect(() => {
+			utils(settings).organizeArguments();
+		}).toThrow();
+	});
+
+	test('Handles having pass-through arguments', () => {
+		const settings = {
+			...settingsPizzaOrdering,
+			arguments: ['order', 'to-go', '--', '--pass-through-flag', 'pass-through-option=true', 'pass-through-data'],
```
Comment (WesCossick) [inline test/utils.test.ts:762]: Same reasoning as above:

```suggestion
			arguments: ['order', 'to-go', '--', '--pass-through-flag', 'pass-through-option=value', 'pass-through-data'],
```

File: test/utils.test.ts [hunk 6]
```diff
@@ -744,6 +744,32 @@ describe('#organizeArguments()', () => {
...
+		};
+
+		expect(utils(settings).organizeArguments()).toStrictEqual({
+			command: 'order to-go',
+			flags: [],
+			options: [],
+			passThrough: ['--pass-through-flag', 'pass-through-option=true', 'pass-through-data'],
```
Comment (WesCossick) [inline test/utils.test.ts:769]: Same reasoning as above:

```suggestion
			passThrough: ['--pass-through-flag', 'pass-through-option=value', 'pass-through-data'],
```

File: src/types.ts (hunk 3 above)
Comment (pbelbin) [reply src/types.ts:39]: Because without the change, TypeScript complains:

Screen Shot 2020-07-27 at 10 23 23 AM
